File: src/main/java/net/lomeli/trophyslots/core/command/CommandUnlockAll.java
package net.lomeli.trophyslots.core.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.ChatComponentText;
import net.minecraft.util.ChatComponentTranslation;
import net.minecraft.util.StatCollector;

import net.lomeli.trophyslots.TrophySlots;
import net.lomeli.trophyslots.core.SlotUtil;

public class CommandUnlockAll extends CommandBase {
    @Override
    public String getCommandName() {
        return "unlock-all";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "command.trophyslots.unlock-all.usage";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (args.length == 1 || args.length == 2) {
            EntityPlayerMP player = args.length == 2 ? getPlayer(sender, args[1]) : getCommandSenderAsPlayer(sender);
            if (player != null) {
                if (SlotUtil.hasUnlockedAllSlots(player))
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.unlock-all.error"), player.getDisplayName())));
                else {
                    TrophySlots.proxy.unlockAllSlots(player);
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.unlock-all.success"), player.getDisplayName())));
                }
            }
        } else
            sender.addChatMessage(new ChatComponentTranslation(getCommandUsage(sender)));
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/command/CommandUnlockAll.kt
package net.lomeli.trophyslots.core.command

import net.lomeli.trophyslots.TrophySlots
import net.lomeli.trophyslots.core.SlotUtil
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.server.MinecraftServer
import net.minecraft.util.text.TextComponentString
import net.minecraft.util.text.TextComponentTranslation
import net.minecraft.util.text.translation.I18n

class CommandUnlockAll : CommandBase() {

    override fun getRequiredPermissionLevel(): Int = 2

    override fun execute(server: MinecraftServer?, sender: ICommandSender?, args: Array<out String>?) {
        if (sender != null) {
            if (args != null && (args.size == 1 || args.size == 2)) {
                val player: EntityPlayerMP
                if (args.size == 2)
                    player = CommandBase.getPlayer(server, sender, args[1])
                else
                    player = CommandBase.getCommandSenderAsPlayer(sender)
                if (player != null) {
                    if (SlotUtil.hasUnlockedAllSlots(player))
                        sender.addChatMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.unlock-all.error").format(player.displayName.unformattedText)));
                    else {
                        TrophySlots.proxy?.unlockAllSlots(player);
                        sender.addChatMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.unlock-all.success").format(player.displayName.unformattedText)));
                    }
                }
            } else
                sender.addChatMessage(TextComponentTranslation(getCommandUsage(sender)));
        }
    }

    override fun getCommandUsage(sender: ICommandSender?): String? = "command.trophyslots.unlock-all.usage"

    override fun getCommandName(): String? = "unlock-all"
}